packages/flutter/test/rendering/sliver_fixed_extent_layout_test.dart

```diff
@@ -148,10 +148,109 @@ void main() {
     expect(parent.paintsChild(children[1]), false);
     expect(parent.paintsChild(children[2]), true);
   });
+
+  test('RenderSliverFillViewport correctly references itemExtent', (){
+    final List<RenderBox> children = <RenderBox>[
+      RenderSizedBox(const Size(400.0, 100.0)),
+      RenderSizedBox(const Size(400.0, 100.0)),
+      RenderSizedBox(const Size(400.0, 100.0)),
+    ];
+    final TestRenderSliverBoxChildManager childManager = TestRenderSliverBoxChildManager(
+      children: children,
+    );
+    final RenderSliverFillViewport sliver = childManager.createRenderSliverFillViewport();
+    final RenderViewport root = RenderViewport(
+      crossAxisDirection: AxisDirection.right,
+      offset: ViewportOffset.zero(),
+      cacheExtent: 100,
+      children: <RenderSliver>[ sliver ],
+    );
+    layout(root);
+    // These are a bogus itemExtents, and motivate the deprecation. The sliver
+    // knows its itemExtent, and should use its configured extent rather than
+    // whatever is provided through these methods.
+    // Also, the API is a bit redundant, so we clean!
+    // In this case, the true item extent is 600 to fill the viewport.
+    expect(
+      sliver.constraints.scrollOffset,
+      0.0
+    );
+    expect(sliver.itemExtent, 600.0);
+    final double layoutOffset = sliver.indexToLayoutOffset(
+      150.0, // itemExtent
+      10,
+    );
+    expect(layoutOffset, 6000.0);
+    final int minIndex = sliver.getMinChildIndexForScrollOffset(
+      sliver.constraints.scrollOffset,
```

Comment:
Using a offset greater than 0 can better test the logic.

Reply:
Very true! Thanks! I will add additional test cases. :)